server/controllers/finance/conventions.js

```diff
@@ -0,0 +1,196 @@
+    ledger             = require('./ledger'),
+    journalConvention  = require('./journal/primarycash').convention;
+
+var getDebitorReceipt  = ledger.debitor_group;
+
+/** create a new convention payment record in the database */
+exports.create = create;
+
+/**
+* POST /cash/conventions
+*
+* @desc This function is responsible for creating a new convention payment record in the database
+* and writing to the posting journal
+*
+* @param {object} PrimaryCashRecord A Primary cash record
+*
+* @example
+* // An example of the object sended in the post request
+* // All object's properties below are required
+* {
+*   uuid            : {uuid},
+*   project_id      : {number},
+*   type            : {char},
+*   date            : {date},
+*   currency_id     : {number},
+*   account_id      : {number},
+*   cost            : {number},
+*   user_id         : {number},
+*   description     : {string},
+*   cash_box_id     : {number},
+*   origin_id       : {number}
+* }
+*/
+function create(req, res, next) {
+  var qData            = req.body,
+      exchangeRate     = {},
+      primaryCashItems = [],
+      overviews        = [];
+
+  var query =
+      'SELECT `debitor_group`.`uuid` FROM `debitor_group` ' +
+      'WHERE `debitor_group`.`account_id`=?';
+
+  db.exec(query, [qData.account_id])
+  .then(getConventionUuid)
+  .then(getDebitorReceipt)
+  .then(getUnpayedReceipts)
+  .then(getExchangeRate)
+  .then(writePay)
+  .then(writeItems)
+  .then(writeToJournal)
+  .catch(next);
+
+  function getConventionUuid(rows) {
+    return rows.pop().uuid;
+  }
+
+  function getUnpayedReceipts(rows) {
+    var situations = rows.filter(function (situation){
+      return situation.balance > 0;
+    });
+    overviews = situations;
+  }
+
+  function getExchangeRate() {
+    var date = util.toMysqlDate(new Date(qData.date)) || util.toMysqlDate(new Date());
+    var query =
+        'SELECT e.id, e.enterprise_currency_id, e.foreign_currency_id, e.rate, e.date FROM exchange_rate e ' +
+        'WHERE e.date = ? ';
+    return db.exec(query, [date])
+    .then(function (rates) {
+      if (!rates.length) { return next(new req.codes.ERR_EXCHANGE_RATE_NOT_FOUND()); }
+      exchangeRate = rates.pop();
+    })
+    .catch(next);
+  }
+
+  function writePay() {
+    var sqlPrimaryCash = 'INSERT INTO primary_cash ' +
+        '(uuid, project_id, type, date, currency_id, account_id, cost, description, cash_box_id, origin_id, user_id) ' +
+        'VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)';
+    var paramPrimaryCash = [
+        qData.uuid, qData.project_id, qData.type, new Date(qData.date), qData.currency_id,
+        qData.account_id, qData.cost, qData.description, qData.cash_box_id, qData.origin_id, qData.user_id
+    ];
+    return db.exec(sqlPrimaryCash, paramPrimaryCash);
+  }
+
+  function writeItems() {
+    primaryCashItems = getPrimaryCashItems(qData.cost, qData.uuid);
+    return q.all(primaryCashItems.map(function (item) {
```

Comment:
You can actually use one statement!  You should be using a transaction, but just to let you know, this query could be written:
```js
return db.exec('INSERT INTO `primary_cash_item` (uuid, col2, col3, etc) VALUES ?;', [ [ primaryCashItems ] ]);
```
You have to give db.exec an array of arrays, but transforms N database queries into a single database query.  Useful for the future.